core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -46,22 +56,43 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.Set;
 import java.util.concurrent.Future;
 
 final class ServerCallImpl<ReqT, RespT> extends ServerCall<RespT> {
   private final ServerStream stream;
   private final MethodDescriptor<ReqT, RespT> method;
   private final Context.CancellableContext context;
+  private final Metadata inboundHeaders;
```

Comment:
Since `Metadata` could be "large," can we save the just parts that are used by this code instead of maintaining a reference to metadata? That would prevent us from maintaining a reference for the entire life of the call.